telemetry/statedata.go

```diff
@@ -0,0 +1,32 @@
+package telemetry
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/hashicorp/go-metrics"
+)
+
+const app_msg_type = "msg_type"
+
+// AppMsgCounter increments a counter with the given data type and success status.
+func AppMsgCounter(ctx context.Context, msgType string, isSuccessful func() bool) {
```

Comment:
What do you think of `EventSuccessCounter`?

I can see this being used for:

- Non-msg-server related things
- App doesn't really add much
- It's always about success or failure of "something"